WcaOnRails/app/controllers/posts_controller.rb

```diff
@@ -73,7 +74,10 @@ def destroy
     #  | 2014 |
     #  +------+
     #  1 row in set, 1 warning (0.00 sec)
-    world_readable_posts = Post.where(world_readable: true)
-    world_readable_posts.find_by_slug(params[:id]) || world_readable_posts.find_by_id!(params[:id])
+    post = Post.find_by_slug(params[:id]) || Post.find_by_id(params[:id])
+    if !post || !post.world_readable
+      raise ActiveRecord::RecordNotFound.new("Couldn't find post")
+    end
+    post
```

Comment:
Weirdly enough, with your suggested query, our tests did pass. I've expanded the relevant test so we won't be tempted to make such a change in the future: https://github.com/jfly/worldcubeassociation.org/commit/fb46ed78739556c62d6b076f2c4cf351d14e1399.